platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/model/internal/monitor/HealthMonitorMetrics.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.wiring.model.internal.monitor;
+
+import static com.swirlds.common.utility.CompareTo.isLessThan;
+
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.metrics.DurationGauge;
+import com.swirlds.metrics.api.IntegerGauge;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+
+/**
+ * Encapsulates metrics for the wiring health monitor.
+ */
+public class HealthMonitorMetrics {
```

Comment:
I like that class since it goes in the direction that I have in mind for custom metrics.